pandas/core/common.py

```diff
@@ -1094,6 +1094,15 @@ def _maybe_promote(dtype, fill_value=np.nan):
             dtype = np.object_
         elif issubclass(dtype.type, (np.integer, np.floating)):
             dtype = np.complex128
+    elif fill_value is None:
+        if issubclass(dtype.type, np.floating):
+            dtype = dtype.type
+            fill_value = np.nan
+        elif issubclass(dtype.type, np.integer):
+            dtype = np.float64
+            fill_value = np.nan
+        else:
+            dtype = np.object_
```

Comment:
Not sure why categorical should raise.  This would seem to be a change in overall functionality of _maybe_promote as previously a categorical would not be promoted as far as I can see.